code/modules/space transition/space_transition.dm

```diff
@@ -74,48 +74,47 @@ var/list/z_levels_list = list()
 		neigbours |= grid[x][y-1]
 
 
-//config/space_levels.txt is where you define your zlevel datum names, their connection to actual z levels and if you want them connected to one another or not
-//Grammar: Name;z value;linked/unlinked
+//zlevel datum names, their connection to actual z levels and if you want them connected to one another or not are defined in each _maps\[map].dm
+//Grammar: [Name].dmm"//[z value] [linked/unlinked]
 //Name is the name of the datum, just for the sake of it
 //z value is to what actual map z level this datum is pointing
 //linked/unlinked decide if you want the z level in the general map or not, for example centcomm is not reachable
-//Each entry must be separated with a single empty line, no spaces outside the name
-//No comments in the file allowed
+//Zlevel amount must be the same as the number of zlevels .dmms included or you'll get a bunch of runtimes
+//MAP_NAME must be the same as the name of the [map].dm file
 
-/proc/setup_map_transitions() //listamania
-	var/list/SLS = file2list("config/space_levels.txt", "\n\n")
+/proc/setup_map_transitions()
+	var/configtext = return_file_text("_maps/[MAP_NAME].dm")
+	var/list/zlist
 	var/datum/space_level/D
-	var/list/config_settings[SLS.len][]
-	for(var/A in SLS)
-		config_settings[SLS.Find(A)] = text2list(A, ";")
-	var/conf_set_len = SLS.len
-	SLS.Cut()
-	for(var/A in config_settings)
+	var/datum/regex/initresults = regex_find(configtext, "zlevel amount:(\\d+)")
+	var/zlen = initresults.str(2)
+	for(var/i, i>zlen, i++)
+		var/datum/regex/results = regex_find(configtext, "(\\w+)\\.dmm\"\\/\\/([i]) (unlinked|linked)")
```

Comment:
What prevernt this code from finding same first line every time?
Do you remove the resulting text from configtext somewhere?